# <font color=red> Start Session </font>

#### Change Kernel to Python

In [1]:
pwd

'C:\\Users\\trini\\Desktop\\Biostatistics\\Week2'

In [2]:
ls

 Volume in drive C is Windows
 Volume Serial Number is 9CFF-211F

 Directory of C:\Users\trini\Desktop\Biostatistics\Week2

09/17/2023  01:13 PM    <DIR>          .
09/16/2023  02:55 PM    <DIR>          ..
09/17/2023  10:32 AM    <DIR>          .ipynb_checkpoints
09/17/2023  01:13 PM            10,663 Biostatistics - Assignment 2.ipynb
09/16/2023  03:01 PM                46 my_data.csv
09/16/2023  03:03 PM               120 my_data.RData
09/16/2023  02:57 PM                30 my_text_file.txt
09/17/2023  10:29 AM           428,120 weight-height.csv
09/17/2023  10:29 AM           168,610 weight-height.csv.gz
               6 File(s)        607,589 bytes
               3 Dir(s)  301,838,868,480 bytes free


In [ ]:
%%bash

# Uncompress the given .csv.gz file from the homework prompt using the BASH terminal
gunzip -k weight-height.csv.gz

#### Change Kernel to R - Start of Weight-height CSV Analysis

In [1]:
# Load csv data file
weight_height_df = read.csv("weight-height.csv", header=TRUE)

In [2]:
# View the CSV file 

head(weight_height_df)
tail(weight_height_df)

Gender Height   Weight  
1 Male   73.84702 241.8936
2 Male   68.78190 162.3105
3 Male   74.11011 212.7409
4 Male   71.73098 220.0425
5 Male   69.88180 206.3498
6 Male   67.25302 152.2122

Gender Height   Weight  
9995  Female 59.09825 110.5297
9996  Female 66.17265 136.7775
9997  Female 67.06715 170.8679
9998  Female 63.86799 128.4753
9999  Female 69.03424 163.8525
10000 Female 61.94425 113.6491

In [3]:
# Get the callable header list:

headers = colnames(weight_height_df)
headers  
# Later if I need to access all data in the header,
# call by weight_height_df$header_name

[1] "Gender" "Height" "Weight"

In [4]:
# Get subset data exclusively only for Male and only for Female

# Subset data only Male Gender
Males_df = weight_height_df[weight_height_df$Gender == "Male",]
# Code above means to call only the Male gender and its entire rows 
# and make it as a subset dataframe
head(Males_df)

# Subset data only Female Gender
Females_df = weight_height_df[weight_height_df$Gender == "Female",]
head(Females_df)

Gender Height   Weight  
1 Male   73.84702 241.8936
2 Male   68.78190 162.3105
3 Male   74.11011 212.7409
4 Male   71.73098 220.0425
5 Male   69.88180 206.3498
6 Male   67.25302 152.2122

Gender Height   Weight  
5001 Female 58.91073 102.0883
5002 Female 65.23001 141.3058
5003 Female 63.36900 131.0414
5004 Female 64.48000 128.1715
5005 Female 61.79310 129.7814
5006 Female 65.96802 156.8021

In [5]:
# Obtain the summary data of the dataframe

summary(weight_height_df)

    Gender         Height          Weight     
 Female:5000   Min.   :54.26   Min.   : 64.7  
 Male  :5000   1st Qu.:63.51   1st Qu.:135.8  
               Median :66.32   Median :161.2  
               Mean   :66.37   Mean   :161.4  
               3rd Qu.:69.17   3rd Qu.:187.2  
               Max.   :79.00   Max.   :270.0  

In [6]:
# Find the mean and std.dev in height and weight for the whole group (Male and Female)
# Hint: Use mean() and sd() function and round of to 2 decimal places.

Male_Female_mean_height = round(mean(weight_height_df$Height),2)
cat ("The mean in height for both Male and Female is:",Male_Female_mean_height, '\n')

Male_Female_stdDev_height = round(sd(weight_height_df$Height),2)
cat ("The std.dev in height for both Male and Female is:",Male_Female_stdDev_height, '\n')

Male_Female_mean_weight = round(mean(weight_height_df$Weight),2)
cat ("The mean in weight for both Male and Female is:",Male_Female_mean_weight, '\n')

Male_Female_stdDev_weight = round(sd(weight_height_df$Weight),2)
cat ("The std.dev in weight for both Male and Female is:",Male_Female_stdDev_weight, '\n')

The mean in height for both Male and Female is: 66.37 
The std.dev in height for both Male and Female is: 3.85 
The mean in weight for both Male and Female is: 161.44 
The std.dev in weight for both Male and Female is: 32.11 


In [7]:
# Find the mean and std.dev in height and weight for Males Only

Males_mean_height = round(mean(Males_df$Height),2)
cat ("The mean in height for Males:",Males_mean_height,'\n')

Males_stdDev_height = round(sd(Males_df$Height),2)
cat ("The std.dev in height for Males:",Males_stdDev_height,'\n')

Males_mean_weight = round(mean(Males_df$Weight),2)
cat ("The mean in weight for Males:",Males_mean_weight,'\n')

Males_stdDev_weight = round(sd(Males_df$Weight),2)
cat ("The std.dev in weight for Males:",Males_stdDev_weight,'\n')

The mean in height for Males: 69.03 
The std.dev in height for Males: 2.86 
The mean in weight for Males: 187.02 
The std.dev in weight for Males: 19.78 


In [8]:
# Find the mean and std.dev in height and weight for Females Only

Females_mean_height = round(mean(Females_df$Height),2)
cat ("The mean in height for Females:",Females_mean_height,'\n')

Females_stdDev_height = round(sd(Females_df$Height),2)
cat ("The std.dev in height for Females:",Females_stdDev_height,'\n')

Females_mean_weight = round(mean(Females_df$Weight),2)
cat ("The mean in weight for Females:",Females_mean_weight,'\n')

Females_stdDev_weight = round(sd(Females_df$Weight),2)
cat ("The std.dev in weight for Females:",Females_stdDev_weight,'\n')

The mean in height for Females: 63.71 
The std.dev in height for Females: 2.7 
The mean in weight for Females: 135.86 
The std.dev in weight for Females: 19.02 


## To determine variability of dataset, 
 Solve for the ***`Coefficient of Variation (CV) = (Std.Dev / Mean) * 100`***
  - This is a dimensionless value in percentage (%) where it `measures the relative variability
      of a dataset` while taking into account
      the relationship of std.dev and mean
  - The `Higher the CV`, the greater the variability around the mean, or the `data values are relatively more widely spread around the mean`
  - The `Lower the CV`, the lesser the variability around the mean, or the `data values are relatively close to the mean`
  

In [9]:
# 5. Solve for the Coefficient of Variation

# I am assuming the homework prompt is being general for all genders in the dataset

CV_height = round(((Male_Female_stdDev_height / Male_Female_mean_height) * 100),2)
cat("The height dataset in both Male and Female has a variability of",CV_height,"% relative to the height mean\n")

CV_weight = round(((Male_Female_stdDev_weight / Male_Female_mean_weight) * 100),2)
cat("The weight dataset in both Male and Female has a variability of",CV_weight,"% relative to the weight mean\n")

# Determine which variability is greater, the prompt said 'Weight'

if (CV_height > CV_weight) {
    print("The variability is greater in Height in both Male and Female dataset")
} else {
    print("The variability is greater in Weight in both Male and Female dataset")
}

### Simpler ifelse I found, take note only
# ifelse(CV_height > CV_weight, "Height", "Weight")


The height dataset in both Male and Female has a variability of 5.8 % relative to the height mean
The weight dataset in both Male and Female has a variability of 19.89 % relative to the weight mean
[1] "The variability is greater in Weight in both Male and Female dataset"


In [10]:
# Create a DataFrame Showing all values calculated
# Easier to view for writer and viewer

df_weight_height = data.frame(
    Mean_height = c(Males_mean_height,Females_mean_height,Male_Female_mean_height),
    StdDev_height = c(Males_stdDev_height, Females_stdDev_height,Male_Female_stdDev_height),
    percent_variability_height = c(NaN,NaN,CV_height),
    Mean_weight = c(Males_mean_weight,Females_mean_weight,Male_Female_mean_weight),
    StdDev_weight = c(Males_stdDev_weight, Females_stdDev_weight,Male_Female_stdDev_weight),
    percent_variability_weight = c(NaN,NaN,CV_weight))

rownames(df_weight_height) = c('Male','Female','Both')
df_weight_height

Mean_height StdDev_height percent_variability_height Mean_weight
Male   69.03       2.86          NaN                        187.02     
Female 63.71       2.70          NaN                        135.86     
Both   66.37       3.85          5.8                        161.44     
       StdDev_weight percent_variability_weight
Male   19.78           NaN                     
Female 19.02           NaN                     
Both   32.11         19.89

#### 6. What can the variability tell you about which variables are more easily influenced?

*Answer: The weight variables are more easily influenced in the variability within the dataset, especially having a standard deviation of 32.11! That is pretty far/skewed from the mean/median*

#### 7. Height and weight have increased over the years. So if you were to go back 50 years, which measurement would you guess has changed most, compared to today's numbers, height or weight (remember you are comparing height and weight which use different units, so what is the best metric to use here ?

*Answer: The weight coefficient of variation measure would comparably change the most because according to Galton 1885, the correlation of heights between generations is not continuous towards an upward nor a downward trend relative to the average height - meaning it remains constant. To elaborate more to my reasoning, weight takes in many factors, but the most crucial factor would be how abundant and accessible the source of food for a population/ecosystem, and that accessibility would also be limited by wealth and health status. In conclusion, the CV of weight measure would have changed the most versus CV of height.*

# <font color=red> End Session </font>